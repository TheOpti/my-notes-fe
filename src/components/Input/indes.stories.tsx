import * as React from 'react';
import { useState } from 'react';
import { storiesOf } from '@storybook/react'

import Input from './'

storiesOf('Input', module)
  .add('basic input with label', () => {
    function Parent({ children, ...props }) {
      const [state, setState] = useState('');
      return <>{children(state, setState)}</>;
    }

    return (
      <Parent>
        {(state, setState) => (
          <Input
            label="Field"
            name="fieldName"
            value={state}
            handleChange={(_, value) => { setState(value)}}
          />
        )}
      </Parent>
    );
  });