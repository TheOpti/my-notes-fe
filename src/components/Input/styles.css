.root {
  position: relative;
  margin: 8px 0;
  max-width: var(--max-form-elem-width);
  width: 100%;
  color: var(--main-dark-gray-color);
}

.input {
  background: none;
  font-size: 18px;
  padding: 10px 10px 10px 5px;
  display: block;
  border: none;
  border-radius: 0;
  border-bottom: 1px solid var(--main-dark-gray-color);
  width: 100%;
}

.input:focus {
  outline: none;
}

.input:focus ~ .label,
.input:valid ~ .label {
  top: -14px;
  font-size: 12px;
  color: var(--main-blue-color);
}

.input:focus ~ .highlight:before {
  width: 100%;
}

.highlight {
  position: relative;
  display: block;
}

.highlight:before {
  content: '';
  height: 2px;
  width: 0;
  bottom: 0px;
  position: absolute;
  background: var(--main-blue-color);
  transition: 0.3s ease all;
  left: 0%;
}

.label {
  font-size: 16px;
  font-weight: normal;
  position: absolute;
  pointer-events: none;
  left: 5px;
  top: 10px;
  transition: 0.3s ease all;
}

.errorMsg {
  height: 16px;
  font-size: 14px;
  margin-top: 8px;
  color: var(--main-red-color);
}
