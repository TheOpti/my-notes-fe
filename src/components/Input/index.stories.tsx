import * as React from 'react';
import { useState } from 'react';
import { storiesOf } from '@storybook/react';

import Input from './';

const btnWrapperStyle = {
	margin: '20px',
	width: '350px',
};

storiesOf('Input', module).add('basic input with label', () => {
	const [state, setState] = useState('');

	return (
		<>
			<div style={btnWrapperStyle}>
				<Input
					label="Field"
					name="fieldName"
					value={state}
					handleChange={(_, value) => {
						setState(value);
					}}
				/>
			</div>
			<div style={btnWrapperStyle}>
				<Input label="Entered value - enabled" name="disabled" value="Entered value" />
			</div>
			<div style={btnWrapperStyle}>
				<Input label="Disabled input" name="disabled" value="Entered value" disabled />
			</div>
		</>
	);
});
