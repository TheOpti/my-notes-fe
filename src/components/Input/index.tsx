import React, { useState } from 'react';
import styles from './styles.css';

type InputPropsType = {
  label: string;
  name: string;
  handleChange?: (event: any, value: string) => void;
  value?: string;
  error?: string;
  formSubmitted?: boolean;
}
const Input: React.SFC<InputPropsType> = (props: InputPropsType) => {
  const { 
    label, name, handleChange,
    value, error, formSubmitted
  } = props;

  const [visited, setVisited] = useState(false);

  const updateFormValue = (event) => {
    const { value } = event.target;
    handleChange(name, value);
  };

  const onBlurHandler = () => {
    setVisited(true);
  };

  const canShowError = visited || formSubmitted;
  
  return (
    <div className={styles.root}>
      <input 
        className={styles.input}
        onChange={updateFormValue}
        onBlur={onBlurHandler}
        value={value}
        required
      />
      <div className={styles.highlight} />
      <label className={styles.label} >
        { label }
      </label>
      <div className={styles.errorMsg}>
        { canShowError && error } 
      </div>
    </div>
  );
}

Input.defaultProps = {
  value: '',
  error: '',
  formSubmitted: false,
}

export default Input;
