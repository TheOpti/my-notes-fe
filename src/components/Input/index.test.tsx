import React from 'react';
import { shallow } from 'enzyme';
import Input from './index';

describe('Input', () => {
  it('should render input with no props without crashing', () => {
    const wrapper = shallow(<Input label="Field" name="fieldName" />);
   
    expect(wrapper).toMatchSnapshot();
  });
});