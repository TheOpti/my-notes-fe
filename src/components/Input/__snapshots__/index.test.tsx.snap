// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render input with no props without crashing 1`] = `
<div
  className="root"
>
  <input
    className="input"
    onBlur={[Function]}
    onChange={[Function]}
    required={true}
    value=""
  />
  <div
    className="highlight"
  />
  <label
    className="label"
  >
    Field
  </label>
  <div
    className="errorMsg"
  />
</div>
`;
